lib/galaxy/files/uris.py

```diff
@@ -51,6 +52,10 @@ def stream_url_to_file(
         with tempfile.NamedTemporaryFile(prefix=prefix, delete=False) as temp:
             temp_name = temp.name
         file_source_path.file_source.realize_to(file_source_path.path, temp_name)
+    elif path.startswith("drs://"):
+        with tempfile.NamedTemporaryFile(prefix=prefix, delete=False) as temp:
+            temp_name = temp.name
+            fetch_drs_to_file(path, temp_name)
```

Comment:
@jmchilton Can this be implemented as a file source instead?  Granted, the file source's list method would return just the initial `source_path` itself, unless it's a collection/bundle, in which case it could return the bundled objects? I ask because as a filesource, it would be far easier to use existing functionality to read auth off the profile etc?

Reply:
I started with the same idea... but to me the abstraction doesn't work because nothing roots ```drs://``` - none of the UI navigation tools would make sense without that? This implementation seems to make more sense - every place we can fetch file sources uses this abstraction anyway so all the cool things we can do with file sources we can do here.

There is some chance I'm being dense about this though - if that would enable vault integration in someway I don't understand maybe sketch a PR or describe the use case and implementation in more detail?

Reply:
The main issue I was wondering about was how auth would be handled with DRS? Specifically, my understanding was that a DRS server may require a [bearer token to gain access](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_making_drs_requests). How would that token be injected here? https://github.com/galaxyproject/galaxy/pull/13949/files#diff-394e589bbc6b016c487d5373c64a4376c6ded4ebb7633d05487a42ca5ac8e218R38

With a filesource, the mechanisms for reading a token from the user profile, vault etc. already exist. Would we have to duplicate that functionality here?

At the risk of stretching the file source abstraction too far, what if all uri were delegated to filesources, including http, base64 and drs? We could add a `supports_root` method to a file source, indicating that it does not support a root listing and can therefore be excluded from the UI?

Reply:
Given the lack of the listing (ls) functionality for DRS, http, etc... - I don't think I'm for this change. Maybe like a parent plugin framework that can handle either file sources or URI fetchers maybe? I think the fact that we've synchronized all URI fetching through this function makes that feasible. 

If you've got a particular test case / use case where that really cleans up things and enables a new DRS source to be used though... I certainly think it is worth considering an implementation.

Reply:
@jmchilton I had a discussion about this with @uwwint and we have a proposal for how file sources can be unified with URI fetchers. I'll create a separate issue to track that.

Reply:
@nuwang Awesome - I look forward to seeing what y'all come up with.